sui_core/src/authority.rs

```diff
@@ -679,12 +679,19 @@ impl AuthorityState {
         signed_effects: SignedTransactionEffects,
     ) -> Result<TransactionInfoResponse, SuiError> {
         let notifier_ticket = self.batch_notifier.ticket()?;
-        self._database.update_state(
+        let signed_effects = self._database.update_state(
             temporary_store,
-            certificate,
-            signed_effects,
+            &certificate,
+            signed_effects.into(),
             Some(notifier_ticket.seq()),
-        )
+        )?;
+        Ok(TransactionInfoResponse {
+            signed_transaction: self
+                ._database
+                .get_signed_transaction(certificate.digest())?,
```

Comment:
Wait, why do we need to get the signed transaction back right away?  Was this moved out of update_state, and if so why move it out of there?